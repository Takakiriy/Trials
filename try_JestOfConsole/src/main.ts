import * as path from "path";  // or path = require("path")
import * as readline from 'readline';
import * as lib from "./lib";

// main
export async function  main() {
    locale = Intl.NumberFormat().resolvedOptions().locale;
    if ('locale' in programOptions) {
        locale = programOptions.locale;
    }

    if ('command' in programOptions) {
        if (programOptions.command === 'stdout') {
            console.log('ABC');
            console.log('DE');
            console.log(programArguments);
        } else if (programOptions.command === 'show-locale') {
            console.log(locale);
        }
    } else if ('input' in programOptions) {
        const  key = await lib.input('input>')
        console.log(key);
    } else {
        console.log('main');
    }
}

function  add(a: number, b: number): number {
    return  a + b;
}


// getStdOut
// Example:
//    var d = getStdOut();  // Set break point here and watch the variable d
function  getStdOut(): string[] {
    return  stdout.split('\n');
}

// pp
// Debug print.
// #keyword: pp
// Example:
//    pp(var);
// Example:
//    var d = pp(var);
//    d = d;  // Set break point here and watch the variable d
// Example:
//    try {
//
//        await main();
//    } finally {
//        var d = pp('');
//        d = [];  // Set break point here and watch the variable d
//    }
export function  pp(message: any) {
    if (typeof message === 'object') {
        message = JSON.stringify(message);
    }
    debugOut.push(message.toString());
    return debugOut;
}
export const  debugOut: string[] = [];

// cc
// Through counter.
// #keyword: cc
// Example:
//   cc();
// Example:
//   var c = cc().debugOut;  // Set break point here and watch the variable c
// Example:
//   if ( cc(2).isTarget )
//   var d = pp('');  // Set break point here and watch the variable d
export function  cc( targetCount: number = 9999999, label: string = '0' ) {
    if (!(label in gCount)) {
        gCount[label] = 0;
    }

    gCount[label] += 1;
    pp( `${label}:countThrough[${label}] = ${gCount[label]}` );
    const isTarget = ( gCount[label] === targetCount );

    if (isTarget) {
        pp( '    **** It is before the target! ****' );
    }
    return  { isTarget, debugOut };
}
const  gCount: {[name: string]: number} = {};

// println
// #keyword: println, console.log, consoleLog
// Output any text to standard output.
export function  println(message: any, delayedExpanding: boolean = false) {
    if (typeof message === 'object' && !delayedExpanding) {
        message = JSON.stringify(message);
    }
    if (withJest && !delayedExpanding) {
        stdout += message.toString() + '\n';
        pp(message.toString());
    } else {
        consoleLog(message);
    }
}
const  consoleLog = console.log;
console.log = println;

// callMainFromJest
// #keyword: callMainFromJest
export async function  callMainFromJest(parameters?: string[], options?: {[name: string]: string}) {
    withJest = true;
    stdout = '';
    if (parameters) {
        programArguments = parameters;
    } else {
        programArguments = [];
    }
    if (options) {
        programOptions = options;
    } else {
        programOptions = {};
    }

    await main();
}

var    locale = '';
var    withJest = false;
export var  stdout = '';
export var  programArguments: string[] = [];
export var  programOptions: {[key: string]: any} = {};
