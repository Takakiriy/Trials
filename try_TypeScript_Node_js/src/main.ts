import * as fs from 'fs';

async function  main() {
	const  exists = fs.existsSync( "_CheckingFile.txt" )
	console.log(exists);
}
main();
