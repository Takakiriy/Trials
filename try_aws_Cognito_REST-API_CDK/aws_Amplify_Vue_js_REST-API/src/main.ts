import Vue from 'vue';
import App from './App.vue';
import router from './router';
import store from './store';
import './registerServiceWorker';

import Amplify, { Logger } from '@aws-amplify/core';
// @ts-ignore
import { AmplifyPlugin } from 'aws-amplify-vue';
import { AmplifyConfig, AmplifyModules, AmplifyVocabularies } from './aws-config';

Amplify.configure(AmplifyConfig);
AmplifyModules.I18n.setLanguage('ja');
AmplifyModules.I18n.putVocabularies(AmplifyVocabularies);
Vue.use(AmplifyPlugin, AmplifyModules);
Vue.config.productionTip = false;

new Vue({
  router,
  store,
  render: (h: any) => h(App),
}).$mount('#app');

Vue.config.errorHandler = (err, vm, info) => {
  const logger = new Logger('AmplifyLogger');
  try {  // for avoiding infinite loop exception
    logger.error( err.stack );
    logger.error( vm.$vnode );
    logger.error( info );
  } catch (e) {
    logger.error( 'in catch' );
  }
};
