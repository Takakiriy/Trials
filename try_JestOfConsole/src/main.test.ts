import * as main from './main';
import * as path from "path";
import * as lib from "./lib";
const  callMain = main.callMainFromJest;

if (path.basename(process.cwd()) !== 'src') {
    // Because the second execute of Jest watch mode is inherited the current folder.
    process.chdir('src');
}

// test
test('First', () => {
    callMain();
});

test('Check stdout', () => {
    callMain(['A', 'B'], {'command': 'stdout'});
    expect(main.stdout).toBe('ABC\nDE\n["A","B"]\n');
});

test('locale', () => {
    const  defaultLocale = Intl.NumberFormat().resolvedOptions().locale;

    callMain([], {'command': 'show-locale'});
    expect(main.stdout).toBe(defaultLocale + '\n');

    callMain([], {'command': 'show-locale', 'locale': 'fr-FR'});
    expect(main.stdout).toBe('fr-FR\n');
});
