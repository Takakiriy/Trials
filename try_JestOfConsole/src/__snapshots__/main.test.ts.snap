// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 >> 1_first_1_ok_1_answer : sourceFileContents 1`] = `
"This is an example of input data.
Hello!
"
`;

exports[`2 >> 2_options_1_command_1_answer : sourceFileContents 1`] = `
"ABC
DE
[\\"A\\",\\"B\\"]
"
`;

exports[`2 >> 2_options_2_input_1_answer : sourceFileContents 1`] = `
"input>Input
"
`;

exports[`snapshot: answer 1`] = `
"This is an example of output data.
Hello!
"
`;
